Source/Csla/Serialization/Mobile/Configuration/MobileFormatterOptions.cs

```diff
@@ -0,0 +1,41 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="MobileFormatterOptions.cs" company="Marimer LLC">
+//     Copyright (c) Marimer LLC. All rights reserved.
+//     Website: https://cslanet.com
+// </copyright>
+// <summary>Options for MobileFormatter configuration</summary>
+//-----------------------------------------------------------------------
+using Csla.Serialization.Mobile;
+
+namespace Csla.Configuration;
+
+/// <summary>
+/// Options for MobileFormatter configuration
+/// </summary>
+public class MobileFormatterOptions
```

Comment:
Somehow I don't think we (should) need this. We should start retrieving the stuff we need from the DI. That makes it a lot easier for user to replace stuff as they deem fit without us to need to think of such scenarios in preparations.
With that said I would suggest to replace the configuration from the current usage (see the tests for the `AddMobileFormatter()` extensions) with something like this:
```csharp
services.AddCsla(o => o.Serialization(
  s => s.AddTypeSerializer<Custom1>().AddTypeSerializer<Custom2>()
            .UseReader<CustomReader>().UseWriter<CustomWriter>()
            .UseSerializer<MyCustomSerializer>()
           [.DontAddDefaultTypeSerializers()?]
)
```
And instead of carrying the implementation type within some options object we should just add everything directly into the DI.
For example: `AddCsla()` adds the `MobileFormatter` as itself _and_ `ISerializationFormatter` implementation. Same goes for the type serializers. We can just add them _all_ into the DI. `MobileFormatter` can retrieve them via `.GetRequiredService<IEnumerable<ITypeMap>>();`.
That would _enable_  user to reuse type map implementations added by us and provide their own.

For the `MobileFormatter` within we _need_ the type. Regardless what the user added to the DI because we have a lot of methods which are dependent on the type instead of it's interface.

Another thought for `AddTypeSerializer` we can have the following signatures:
* `AddTypeSerializer<T>() where T: TypeMap<T, S>`
* `AddTypeSerializer<T, S>() where S: IMobileSerializer`
* ...

Reply:
It is gone